src/lambda/config/templates.ts

```diff
@@ -25,49 +24,54 @@ export interface TemplateConfig {
 }
 
 export interface LoadTemplatesConfigContext {
-    logger: Pick<Logger, 'warn'>
-    readFile(path: string): Promise<string>
+    fileExists(path: string): Thenable<boolean>
+    readFile(path: string): Thenable<string>
+    saveDocumentIfDirty(editorPath: string): Thenable<void>
 }
 
-function getWorkspaceFolderPath(input: Pick<vscode.WorkspaceFolder, 'uri'> | vscode.Uri | string): string {
-    if (typeof input === 'string') {
-        return input
-    }
+export class DefaultLoadTemplatesConfigContext implements LoadTemplatesConfigContext {
+    public readonly fileExists = fsUtils.fileExists
+    public readonly readFile = fsUtils.readFileAsString
 
-    const workspaceFolder = input as Pick<vscode.WorkspaceFolder, 'uri'>
-    if (workspaceFolder.uri) {
-        return workspaceFolder.uri.fsPath
-    }
+    public async saveDocumentIfDirty(editorPath: string): Promise<void> {
+        const path = _path.normalize(vscode.Uri.file(editorPath).fsPath)
+        const document = vscode.workspace.textDocuments.find(doc => {
+            if (doc.isDirty) {
```

Comment:
This lambda returns documents that "are not dirty, and match editorPath".
Did you mean documents that "are editor path and are dirty"?

Reply:
✔️